dev/AccessControl/GetSecurityDescriptorForAppContainerNames.cpp

```diff
@@ -0,0 +1,100 @@
+﻿#include <pch.h>
+#include <wil/resource.h>
+#include <wil/result.h>
+#include <accctrl.h>
+#include <aclapi.h>
+#include <userenv.h>
+#include "Security.AccessControl.h"
+#include "SecurityDescriptorHelpers.h"
+#include <vector>
+
+using namespace std;
+
+STDAPI GetSecurityDescriptorForAppContainerNames(
+    uint32_t accessRequestCount,
+    _In_reads_(accessRequestCount)
+    const AppContainerNameAndAccess* accessRequests,
+    _In_opt_ PSID principal,
+    uint32_t principalAccessMask,
+    _Outptr_ PSECURITY_DESCRIPTOR* securityDescriptor,
+    _Out_opt_ DWORD* securityDescriptorLength
+)
+{
+    *securityDescriptor = nullptr;
+    if (securityDescriptorLength)
+    {
+        *securityDescriptorLength = 0;
+    }
+    try
+    {
+        vector<EXPLICIT_ACCESS_W> ea;
+        if (accessRequestCount == UINT32_MAX)
+        {
+            THROW_HR(E_OUTOFMEMORY);
+        }
+        ea.reserve(static_cast<size_t>(accessRequestCount) + 1);
```

Comment:
Leftover code trying to be resilient to overflow. If `accessRequestCount == UINT32_MAX`, then adding one would overflow before the promotion to 64-bit `size_`t.

Ended up simply throwing out of the function to avoid this case, but this little nugget was left behind.